crates/astria-sequencer-relayer/src/relayer/mod.rs

```diff
@@ -288,6 +303,63 @@ impl Relayer {
     }
 }
 
+#[instrument(skip_all)]
+async fn confirm_sequencer_chain_id(
+    configured_sequencer_chain_id: String,
+    sequencer_cometbft_client: SequencerClient,
+) -> eyre::Result<()> {
+    let span = Span::current();
+
+    let retry_config = tryhard::RetryFutureConfig::new(u32::MAX)
+        .max_delay(Duration::from_secs(30))
+        .exponential_backoff(Duration::from_secs(1))
+        .on_retry(
+            |attempt: u32, next_delay: Option<Duration>, error: &tendermint_rpc::Error| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    parent: &span,
+                    attempt,
+                    wait_duration,
+                    error = %eyre::Report::new(error.clone()),
+                    "failed to fetch sequencer chain id; retrying after backoff",
+                );
+                futures::future::ready(())
+            },
+        );
+
+    let actual_sequencer_chain_id =
+        tryhard::retry_fn(move || fetch_sequencer_chain_id(sequencer_cometbft_client.clone()))
+            .with_config(retry_config)
+            .in_current_span()
+            .await
+            .wrap_err("retry attempts exhausted; bailing")?;
+
+    if actual_sequencer_chain_id == configured_sequencer_chain_id {
+        info!(sequencer_chain_id = %configured_sequencer_chain_id, "confirmed sequencer chain id");
+        return Ok(());
+    }
+    bail!(
+        "mismatch in sequencer chain id, configured id: `{configured_sequencer_chain_id}`, actual \
+         id: `{actual_sequencer_chain_id}`"
+    )
+}
+
+async fn fetch_sequencer_chain_id(
+    sequencer_cometbft_client: SequencerClient,
+) -> Result<String, tendermint_rpc::Error> {
+    use sequencer_client::Client as _;
+
+    let response = sequencer_cometbft_client.status().await;
+    // trace-level logging, so using Debug format is ok.
+    #[cfg_attr(dylint_lib = "tracing_debug_field", allow(tracing_debug_field))]
+    {
+        trace!(?response);
+    }
+    response.map(|status_response| status_response.node_info.network.to_string())
```

Comment:
the fact that this var is called `network` and not `chain_id` as in other rpc responses (such as genesis and block) bothers me a bit :p I think it's fine to leave though, as it should be the same as the chain id

Reply:
Yeah - it doesn't sit well with me either, but it's part of the tendermint API, so my hands are pretty much tied on this one I think.